mlir/lib/Dialect/Rock/IR/RockDialect.cpp

```diff
@@ -663,6 +670,11 @@ LogicalResult GemmOp::verify() {
     return emitOpError("general kernels don't support non-set store methods");
   }
 
+  if (getDerivedBlockSize().has_value() && !isXdlops) {
```

Comment:
Again, slightly not the right validation. `getDerivedBlockSize().has_value() && hasGeneralGemmParams`
where `hasGeneralGemmParams = getParams.isa_and_nonnull<General...ParamsAttr>()`

Reply:
OK, just understood what you mean, it also can be forced to set. That makes better sense.

Reply:
I mean that the condition is not "we set derived block size on a non-xdlops kernel", it's "we set both the derived block size and included tuning parameters that contain a block size"

Because non-xdlops utility kernels will set the derived block size and delete the tuning parameters

Reply:
That reminds me I reverted deleting params attr but found correct place to delete it.

Reply:
Wait, this is still correct. An utility kernel itself can't be a gemmOp, it's lowered in `convToGemm`. Correct derivedBlockSize is used for the actual gemmOp, not utilityBlockSize.

Reply:
I want the if statement to test what we actually care about, not a proxy for it.

The test is "is it the case that we have set both the derived block size and set a tuning parameters attribute that specified a block size". `isXdlops` has nothing to do with it.

Reply:
Right, that could be a possible error case, added it.
I wanted to check the case derived blocksize attr is set in xdlops path, which is a separate case.

Reply:
Yeah, fair